crates/aggchain-proof-core/src/bridge/mod.rs

```diff
@@ -424,7 +449,49 @@ fn filter_values<I: Eq + Hash + Copy>(removed: &[I], values: &[I]) -> Vec<I> {
             true // Otherwise, keep the value.
         })
         .copied()
-        .collect()
+        .collect();
+
+    Ok(result)
+}
+
+fn filter_bridge_exits(
+    removed_indices: &[B256],
+    exits: &[BridgeExit],
+) -> Result<Vec<BridgeExit>, BridgeConstraintsError> {
+    // Create a map that counts how many removals are needed for each index
+    let mut removal_map: HashMap<B256, usize> = HashMap::new();
+    for &value in removed_indices {
+        let count = removal_map.entry(value).or_insert(0);
+        *count = count
+            .checked_add(1)
+            .ok_or(BridgeConstraintsError::HashChainOverflow)?;
+    }
+
+    // Filter exits based on their global_index
+    let result = exits
+        .iter()
+        .filter(|exit| {
+            // If this exit's global_index needs to be removed...
+            if let Some(count) = removal_map.get_mut(&exit.global_index) {
+                if *count > 0 {
+                    *count -= 1; // Remove one occurrence
+                    return false; // Skip this exit
+                }
+            }
+            true // Keep this exit
+        })
+        .cloned() // Since we're working with references, clone to own the data
+        .collect();
+
+    Ok(result)
+}
+
+pub fn compute_commit_imported_bridge_exits(bridge_exits_claimed: Vec<BridgeExit>) -> Digest {
```

Comment:
The function name sound strange. How about this?
```suggestion
pub fn compute_imported_bridge_exits_commitment(bridge_exits_claimed: Vec<BridgeExit>) -> Digest {
```